google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/GCRules.java

```diff
@@ -0,0 +1,315 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.admin.v2.models;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nonnull;
+import org.threeten.bp.Duration;
+import com.google.api.core.BetaApi;
+import com.google.api.core.InternalApi;
+import com.google.bigtable.admin.v2.GcRule;
+import com.google.bigtable.admin.v2.GcRule.Intersection;
+import com.google.bigtable.admin.v2.GcRule.Union;
+import com.google.common.base.MoreObjects;
+
+/** Wraps {@link GcRule} protocol buffer object and exposes a simpler Fluent DSL model */
+@BetaApi
+public final class GCRules {
+  /** Factory method to create GCRules - entry point into the DSL. */
+  public static final GCRules GCRULES = new GCRules();
+
+  private GCRules() {}
+
+  /**
+   * Creates a new instance of the IntersectionRule
+   *
+   * @return IntersectionRule
+   */
+  public IntersectionRule intersection() {
+    return new IntersectionRule();
+  }
+
+  /**
+   * Creates a new instance of the IntersectionRule
+   *
+   * @return UnionRule
+   */
+  public UnionRule union() {
+    return new UnionRule();
+  }
+
+  /**
+   * Creates a new instance of the VersionRule
+   *
+   * @param maxVersion - maximum number of cell versions to keep
+   * @return VersionRule
+   */
+  public VersionRule maxVersions(int maxVersion) {
+    return new VersionRule(maxVersion);
+  }
+
+  /**
+   * Creates a new instance of the DurationRule
+   *
+   * @param maxAge - maximum age of the cell to keep
+   * @param timeUnit - timeunit for the age
+   * @return DurationRule
+   */
+  public DurationRule maxAge(long maxAge, TimeUnit timeUnit) {
+    Duration duration = Duration.ZERO;
+    TimeUnit.SECONDS.convert(maxAge, timeUnit);
+    return maxAge(duration);
+  }
+
+  /**
+   * Creates a new instance of the DurationRule
+   *
+   * @param duration - age expressed as duration
+   * @return DurationRule
+   */
+  public DurationRule maxAge(Duration duration) {
+    return new DurationRule(duration);
+  }
+
+  /**
+   * Creates an empty default rule
+   *
+   * @return DefaultRule
+   */
+  public DefaultRule defaulRule() {
+    return new DefaultRule();
+  }
+
+  /**
+   * Fluent wrapper for {@link Intersection} rule. Allows far adding an hierarchy of rules with
+   * intersection as the root
+   */
+  public static final class IntersectionRule extends BaseRule {
+    private GcRule.Intersection.Builder builder;
+    private List<GCRule> rulesList = new ArrayList<>();
```

Comment:
I think it would be better to only store rulesList and to build the proto in toProto on the fly.